flaml/model.py

```diff
@@ -736,27 +734,22 @@ def _select_checkpoint(self, trainer):
     def _compute_metrics_by_dataset_name(self, eval_pred):
         if isinstance(self._metric, str):
             from .ml import metric_loss_score
-            from .nlp.utils import postprocess_text
+            from .nlp.utils import (
+                postprocess_prediction,
+                postprocess_seq2seq_prediction_label,
+            )
 
             predictions, labels = eval_pred
+            predictions = postprocess_prediction(
+                self._task, predictions, self.tokenizer
+            )
             if self._task in NLG_TASKS:
-                if isinstance(predictions, tuple):
-                    predictions = np.argmax(predictions[0], axis=2)
-                decoded_preds = self.tokenizer.batch_decode(
-                    predictions, skip_special_tokens=True
-                )
                 labels = np.where(labels != -100, labels, self.tokenizer.pad_token_id)
                 decoded_labels = self.tokenizer.batch_decode(
                     labels, skip_special_tokens=True
                 )
-                predictions, labels = postprocess_text(decoded_preds, decoded_labels)
-            else:
-                predictions = (
-                    np.squeeze(predictions)
-                    if self._task == SEQREGRESSION
-                    else np.argmax(predictions, axis=2)
-                    if self._task == TOKENCLASSIFICATION
-                    else np.argmax(predictions, axis=1)
+                predictions, labels = postprocess_seq2seq_prediction_label(
+                    predictions, decoded_labels
                 )
             metric_dict = {
                 "automl_metric": metric_loss_score(

```

Comment:
Could you clarify what is expected as input to:
1. metric_loss_score? For example, it's not clear to just say it expects `y_predict` and `y_true`. What processing is required for `y_predict` and `y_true`.
2. self._metric? For example, what processing is required for y_test?
3. TransformerEstimator.fit? For example, what processing is required for y_val?
Please add necessary info in docstr to clarify these, and check whether your current code satisfies the requirements.
Right now, I don't understand the requirements and can't check the correctness of the code.